bazel/repositories.bzl

```diff
@@ -241,7 +241,6 @@ def _com_github_c_ares_c_ares():
     location = _get_location("com_github_c_ares_c_ares")
     http_archive(
         name = "com_github_c_ares_c_ares",
-        patches = ["@envoy//bazel/foreign_cc:cares-win32-nameser.patch"],
```

Comment:
Question, why was this dropped? The //source/exe:envoy-static doesn't require this patch, however, the //test/common/network:dns_impl_test does need this hack to provide windows a nameser.h (which isn't in the windows API at all, that's why c-ares bundles the file.)

Reply:
@wrowe is this in the tests that are exercised by CI right now? The Windows job is passing. I dropped the patch as I wasn't sure it was needed and didn't want to do the work of updating if not; in general these patches are going to impact our ability to upgrade versions if they aren't upstreamed.

Reply:
Correct, right now we don't build tests on Windows, because the Azure workers don't have sufficient memory to compile much of anything with respect to the mocks (they eat up about 4GB of heap when included.) But it is necessary to retain that patch, and it won't be absorbed upstream because of what we are accomplishing with it. We are "borrowing" their nameser.h, because it is the most convenient way to obtain it.

Reply:
(Sunjay and I are building the tests frequently, and they are going live with the move to the RBE picture for Windows builds.)

Reply:
I think there might be other ways to get this header, e.g. via a `genrule` that extracts it from the `http_archive`. Or maybe even directly via a repository rule. I'd be happy to work on that in this PR if there was a way to validate it. Is there a way to enable this test? 

I'm very much opposed to permanently maintaining cmake patches, since these are fragile and will impede our ability to track security updates by imposing a higher cost on folks making these updates.

Reply:
While I don't disagree with you, we can simply disable the test with tags = ["skip_on_windows"] and just forget about it. Looking at the situation, this seemed at the time to be the most viable.